### PR TITLE
Issue87 star behaviour and swiping

### LINKED ISSUE #87: UX/UI - Star behaviour + swiping
Currently users can tap a faded star icon in order to ... do something. 
Tapping the icon does not seem to do anything
Tapping the icon more than once returns an error 'duplicate record exists'
It is difficult to tap the star icon without tapping the event image which takes you to the Event details page
There is no way to see events that you have starred (what is the point of this feature?)
**Proposed Enhancements:**
surface a way to view starred event
swap faded star icon for a 'swipe left to star' behaviour. This will reinforce the concept of swipe actions and navigation throughout the application (see ticket #775) This also solves contrast issues with the faded star icon not being visible on certain background images. 
this would require a few UI updates to make sure that swipe actions on an Event listing is intuitive. Mainly, separating event listings with a light border would emphasise each listing as its own row and would set a comfortable boundary around the information that is swipable.

### PR DESCRIPTION
**Pull request by mrmaDrma**
Resolves #87 

I've added state, that handles active/unactive star.

@Krakaw @Tas4Tari
Now there are 2 problems I've found.
For some reason whenever I trigger api for handling favorite event, I get `events is expecting these fields: name, organization_id, event_type` in my console.

Another problem is that after I rerun application, and get all events again. Every event has `user_is_interested` property set to false, even if I marked is as true.
I would like someone to check this api and work together with me to find if issue is on my side or on apis?

### CODE REVIEW

**Comment by Krakaw** [comment]
I will look into the API and get back to you. Please mark this issue as either **[DNM]** or **[WIP]** if it is not ready to be merged in.

File: src/state/eventStateProvider.js [hunk 1]
```diff
@@ -275,13 +275,13 @@ class EventsContainer extends Container {
   // allEvents will refresh all events (ie: from the index page), whereas setting it to false will refresh the interested event
   toggleInterest = async (event, singleEvent = false) => {
     const {user_is_interested, id} = event
+    const {remove, create} = server.events
```
Comment (Krakaw) [inline src/state/eventStateProvider.js]: This is the wrong endpoint, you are trying to create and actual event not toggling the favorite.
```suggestion
    const {remove, create} = server.events.interesets
```

**Comment by Krakaw** [comment]
I applied the fix locally to test and it works, but on the index page the star image state is not updating. http://cl.ly/98fe6f962b87

**Comment by mileTadic1994** [comment]
> I applied the fix locally to test and it works, but on the index page the star image state is not updating. http://cl.ly/98fe6f962b87

Oh sorry, I've missed that. Will fix it today.

**Comment by Krakaw** [comment]
> > I applied the fix locally to test and it works, but on the index page the star image state is not updating. http://cl.ly/98fe6f962b87
> 
> Oh sorry, I've missed that. Will fix it today.

Is this fixed?

**Comment by mileTadic1994** [comment]
> > > I applied the fix locally to test and it works, but on the index page the star image state is not updating. http://cl.ly/98fe6f962b87
> > 
> > 
> > Oh sorry, I've missed that. Will fix it today.
> 
> Is this fixed?

I'm still working on this. The problem is I cannot get new events list when user goes back from details page. So I'm trying to find the solution for this problem. I tried a lot of different things yesterday, but it didn't work. I have one more solution that I want to try, if it doesn't work, I will have to talk with @simple-taskic if he has any solution.

**Comment by Krakaw** [comment]
Is this ready?

**Comment by mileTadic1994** [comment]
> Is this ready?

Not yet, still has a problem when starring event from details event page and then going back to all events page.

**Pull request merged**